src/tasks/tasks.proj

```diff
@@ -1,12 +1,27 @@
 <Project Sdk="Microsoft.Build.Traversal">
   <ItemGroup>
-    <ProjectReference Include="$(MSBuildThisFileDirectory)**\*.csproj" />
+    <MonoTaskProject Include="AndroidAppBuilder\AndroidAppBuilder.csproj" />
+    <MonoTaskProject Include="AotCompilerTask\MonoAOTCompiler.csproj" />
+    <MonoTaskProject Include="AppleAppBuilder\AppleAppBuilder.csproj" />
```

Comment:
we do have NativeAOT support on Apple/WASM so I don't think grouping these as `MonoTaskProject` makes sense.
I'd rather do the split by OS, but I'm not sure if the workloads build throws a wrench into that. @steveisok

Reply:
A TargetsMobile check might be appropriate here. @akoeplinger that might satisfy the official build issue we talked about yesterday.

Reply:
I do agree the item name shouldn't contain mono.

Reply:
> A TargetsMobile check might be appropriate here. @akoeplinger that might satisfy the official build issue we talked about yesterday.

FWIW there is a new official build for 8.0.8 which ran yesterday that did not have a retry. The produced MonoTargets.Sdk packages and the workload contents are matching

Reply:
@ViktorHofer @akoeplinger  tasks.proj seems to run too early to pick up `TargetsMobile` that's defined in the top level `Directory.Build.targets`. Any suggestions besides duplicating?

Reply:
Hm, I tried the build locally and `TargetsMobile` was properly set before tasks.proj. Which scenario was failing so that we need to duplicate it?

Reply:
From an msbuild evaluation order perspective, the `TargetsMobile` property should be available in tasks.proj.

Reply:
> From an msbuild evaluation order perspective, the `TargetsMobile` property should be available in tasks.proj.

For whatever reason, it's not available. I can't tell why.

Reply:
Can you please share a binlog offline that demonstrates that? I just looked at an ios binlog from a runtime official build and I see the property being defined:

![image](https://github.com/dotnet/runtime/assets/7412651/73bcd337-756c-4355-a816-74f7dbe92a4d)

Reply:
From the CI result:

![image](https://github.com/dotnet/runtime/assets/7412651/fcef9fe3-ee25-43af-a198-5065097fd718)

And then looking at "Evaluations" -> tasks.proj -> Properties:

![image](https://github.com/dotnet/runtime/assets/7412651/d88964f1-576d-43ad-b201-47e8f6d4bf2b)

Reply:
Sigh, maybe the difference is that it's a new day ;-). Glad we don't need to do something else then.